packages/trader/src/AppV2/Containers/Trade/trade-types.tsx

```diff
@@ -1,28 +1,230 @@
+        const saved_pinned_trade_types = localStorage.getItem('pinned_trade_types');
+
+        onMount();
+
+        if (saved_pinned_trade_types) {
+            setPinnedTradeTypes(JSON.parse(saved_pinned_trade_types));
+        }
+
+        if (saved_other_trade_types) {
+            setOtherTradeTypes(JSON.parse(saved_other_trade_types));
+        }
+
+        return () => {
+            onUnmount();
+        };
+    }, []);
+
+    const savePinnedToLocalStorage = () => {
+        localStorage.setItem('pinned_trade_types', JSON.stringify(pinned_trade_types));
+        localStorage.setItem('other_trade_types', JSON.stringify(other_trade_types));
+        setIsOpen(false);
+    };
+
+    const handleOnDrag = (categories: TResultItem[]) => {
+        setPinnedTradeTypes(categories);
+    };
+
     const isTradeTypeSelected = (value: string) =>
         [contract_type, value].every(type => type.startsWith('vanilla')) ||
         [contract_type, value].every(type => type.startsWith('turbos')) ||
         [contract_type, value].every(type => type.startsWith('rise_fall')) ||
         contract_type === value;
     return (
         <div className='trade__trade-types'>
-            {trade_types.map(({ text, value }) => (
-                <Chip.Selectable key={value} onChipSelect={onTradeTypeSelect} selected={isTradeTypeSelected(value)}>
-                    <Text size='sm'>{text}</Text>
-                </Chip.Selectable>
-            ))}
+            {saved_pinned_trade_types &&
+                JSON.parse(saved_pinned_trade_types)[0].items.map(({ title, id }: TItem) => (
+                    <Chip.Selectable key={id} onChipSelect={onTradeTypeSelect} selected={isTradeTypeSelected(id)}>
+                        <Text size='sm'>{title}</Text>
+                    </Chip.Selectable>
+                ))}
+            <a key={'all'} onClick={() => setIsOpen(true)} className='trade__trade-types-header'>
+                <Text size='sm' bold underlined>{<Localize i18n_default_text='View all' />}</Text>
+            </a>
+            <ActionSheet.Root isOpen={is_open} expandable={false} onClose={handleCloseTradeTypes}>
+                <ActionSheet.Portal>
+                    <ActionSheet.Header title={<Localize i18n_default_text='Trade types' />} />
+                    <ActionSheet.Content className='mock-action-sheet--content'>
+                        {is_editing ? (
+                            <DraggableList
+                                categories={pinned_trade_types}
+                                onRightIconClick={handleRemovePinnedClick}
+                                onAction={savePinnedToLocalStorage}
+                                onDrag={handleOnDrag}
+                            />
+                        ) : (
+                            <TradeTypeList
+                                categories={saved_pinned_trade_types && JSON.parse(saved_pinned_trade_types)}
```

Comment:
JSON.parse(saved_pinned_trade_types) is used in two places in this render. Can we set it to a variable and use it in the render?